pkg/controller/config/config_controller.go

```diff
@@ -483,73 +483,44 @@ func (r *ReconcileConfig) serviceAccountNameForDeployment(deployment types.Names
-	privileged, err := r.secClient.SecurityV1().SecurityContextConstraints().Get("privileged", metav1.GetOptions{})
-	if err != nil {
-		log.Error(err, "scc privileged get error")
+	// get the scc and not nothing if it exists
+	scc := r.secClient.SecurityV1().SecurityContextConstraints()
+	sccPriv, err := scc.Get(flag.OperatorPrivilegedSCC, metav1.GetOptions{})
+	if err == nil || !errors.IsNotFound(err) {
+		log.Error(err, "failed to find scc")
 		return err
 	}
 
-	newList, changed := addToList(privileged.Users, sa)
-	_, annotated := privileged.Annotations[flag.SccAnnotationKey]
-	if !changed && annotated {
-		log.Info("scc already in added to the list", "action", "none")
-		return nil
-	}
-
-	log.Info("privileged scc needs updation")
-	privileged.Annotations[flag.SccAnnotationKey] = sa
-	privileged.Users = newList
-
-	updated, err := r.secClient.SecurityV1().SecurityContextConstraints().Update(privileged)
-	log.Info("added SA to scc", "updated", updated.Users)
-	return err
-}
+	log.Info("err ", "whaaaat", err)
 
-func (r *ReconcileConfig) removePrivilegedSCC() error {
-	log := ctrlLog.WithName("scc").WithName("remove")
-	privileged, err := r.secClient.SecurityV1().SecurityContextConstraints().Get("privileged", metav1.GetOptions{})
+	// craete new copied for privileged
+	privileged, err := scc.Get("privileged", metav1.GetOptions{})
 	if err != nil {
 		log.Error(err, "scc privileged get error")
 		return err
 	}
+	log.Info("creating scc based on privileged")
 
-	sa, annotated := privileged.Annotations[flag.SccAnnotationKey]
-	if !annotated {
-		log.Info("sa already not in privileged SCC", "action", "none")
-		return nil
-	}
-
-	newList, changed := removeFromList(privileged.Users, sa)
-	if !changed {
-		log.Info("sa already not in privileged SCC", "action", "none")
-		return nil
-	}
-
-	log.Info("privileged scc needs updation")
-	delete(privileged.Annotations, flag.SccAnnotationKey)
-	privileged.Users = newList
+	sccPriv = privileged.DeepCopy()
+	sccPriv.ObjectMeta = metav1.ObjectMeta{Name: flag.OperatorPrivilegedSCC}
+	sccPriv.Users = []string{sa}
 
-	updated, err := r.secClient.SecurityV1().SecurityContextConstraints().Update(privileged)
-	log.Info("removed SA from scc", "updated", updated.Users)
+	created, err := scc.Create(sccPriv)
+	log.Info("Created new SCC based on privileged", "users", created.Users)
 	return err
 }
 
-func removeFromList(list []string, item string) ([]string, bool) {
-	for i, v := range list {
-		if v == item {
-			return append(list[:i], list[i+1:]...), true
-		}
-	}
-	return list, false
-}
+func (r *ReconcileConfig) removePrivilegedSCC() error {
+	log := ctrlLog.WithName("scc").WithName("remove")
+	scc := r.secClient.SecurityV1().SecurityContextConstraints()
 
-func addToList(list []string, item string) ([]string, bool) {
-	for _, v := range list {
-		if v == item {
-			return list, false
-		}
+	if err := scc.Delete(flag.OperatorPrivilegedSCC, &metav1.DeleteOptions{}); err != nil {
```

Comment:
nice, now we can just delete, if present. instead of having to do a remove item and patch